chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/UserTestingLabMapDAOImpl.java

```diff
@@ -0,0 +1,163 @@
+
+        List<UserTestingLabMapDTO> dtos = new ArrayList<UserTestingLabMapDTO>();
+        if (result != null) {
+            for (UserTestingLabMapEntity entity : result) {
+                dtos.add(new UserTestingLabMapDTO(entity));
+            }
+        }
+        return dtos;
+    }
+
+    @Override
+    public List<UserTestingLabMapDTO> getByAtlId(Long atlId) {
+        Query query = entityManager.createQuery(
+                "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab tl "
+                        + "join fetch utlm.user u where (utlm.deleted != true) AND (tl.id = :atlId) ",
+                UserTestingLabMapEntity.class);
+        query.setParameter("atlId", atlId);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        List<UserTestingLabMapDTO> dtos = new ArrayList<UserTestingLabMapDTO>();
+        for (UserTestingLabMapEntity entity : result) {
+            dtos.add(new UserTestingLabMapDTO(entity));
+        }
+        return dtos;
+    }
+
+    @Override
+    public UserTestingLabMapDTO getById(Long id) {
+        Query query = entityManager.createQuery(
+                "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab TestingLabEntity "
+                        + "join fetch utlm.user UserEntity" + "where (utlm.deleted != true) AND (utlm.id = :id) ",
+                UserTestingLabMapEntity.class);
+        query.setParameter("id", id);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        if (result.size() == 0) {
+            return null;
+        }
+        return new UserTestingLabMapDTO(result.get(0));
+    }
+
+    private UserTestingLabMapEntity create(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        if (entity.getRetired() == null) {
+            entity.setRetired(false);
+        }
+        if (entity.getDeleted() == null) {
+            entity.setDeleted(false);
+        }
+        entityManager.persist(entity);
+        return entity;
+    }
+
+    private UserTestingLabMapEntity update(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        entityManager.persist(entity);
+        return entity;
+    }
+
+    private TestingLabEntity getAtlEntityById(final Long entityId) throws EntityRetrievalException {
+        TestingLabEntity entity = null;
+
+        String queryStr = "SELECT atl from TestingLabEntity atl " + "LEFT OUTER JOIN FETCH atl.address "
+                + "WHERE (atl.id = :entityid)" + " AND (atl.deleted = false)";
+
+        Query query = entityManager.createQuery(queryStr, TestingLabEntity.class);
+        query.setParameter("entityid", entityId);
+        List<TestingLabEntity> result = query.getResultList();
+
+        if (result == null || result.size() == 0) {
+            String msg = String.format(messageSource.getMessage(new DefaultMessageSourceResolvable("acb.notFound"),
```

Comment:
Could you use the ErrorMessageUtil here? Also should the message be something about atl not found instead of acb.notFound?